### PR TITLE
feat: add `<Banner>` component

### PR DESCRIPTION
**Pull request by tulup-conner**
- [x] I have followed the [Your First Code Contribution section of the Contributing guide](https://github.com/themesberg/flowbite-react/blob/main/CONTRIBUTING.md#your-first-code-contribution)

Adds the [Banner](https://flowbite.com/docs/components/banner/) component from the main Flowbite library.

This component is currently a work-in-progress; you can expect that there may be significant changes in the future, in particular, to its theme. There currently isn't a theme, but there probably will be in the future.

Unit tests and Storybook stories will be added in the future.

### CODE REVIEW

File: src/components/Banner/Banner.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import type { ComponentProps, FC } from 'react';
+import { BannerCollapseButton } from './BannerCollapseButton';
+
+export type BannerComponentProps = ComponentProps<'div'>;
+
+const BannerComponent: FC<BannerComponentProps> = ({ children, ...props }) => {
+  return (
+    <div data-testid="flowbite-banner" tabIndex={-1} {...props}>
```
Comment (nigellima) [inline src/components/Banner/Banner.tsx:8]: Add a `role="banner"` here for screen readers

Comment (tulup-conner) [reply src/components/Banner/Banner.tsx:8]: Thank you, doing this now. Also, this isn't in the vanilla library @zoltanszogyenyi

File: src/components/Banner/BannerCollapseButton.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import type { FC, MouseEventHandler } from 'react';
+import type { ButtonProps } from '../Button';
+import { Button } from '../Button';
+
+export type BannerCollapseButtonProps = ButtonProps;
+
+export const BannerCollapseButton: FC<BannerCollapseButtonProps> = ({ children, ...props }) => {
+  const onClick: MouseEventHandler<HTMLButtonElement> = (event) => {
+    const collapseButton = event.target as HTMLButtonElement;
+    const parentBanner = collapseButton.closest('[data-testid="flowbite-banner"]');
```
Comment (nigellima) [inline src/components/Banner/BannerCollapseButton.tsx:10]: I'm not sure if using the data-testid here for implementation logic is advisable.

Maybe use the `role=banner` here to identify the parent

Comment (tulup-conner) [reply src/components/Banner/BannerCollapseButton.tsx:10]: Agree now that we have that, will change that as well now.

**Pull request merged**